pyproject.toml

```diff
@@ -134,6 +135,33 @@ if [ $(echo "$output" | wc -l) -gt 1 ]; then
 echo "$output"; exit 1;
 fi
 """
+# Run check on the sbom to see if it needs to be updated.
+# We create a new sbom from the existing source for the almanack,
+# then compare it to the current sbom.json for differences.
+# If the sbom is missing or differs, we throw a non-zero exit.
+# Intended to be run through pre-commmit as a check on this repo.
+sbom-checks.shell = """
+# Generate SBOM using Anchore Syft
+docker run --rm -v $(pwd):/workspace anchore/syft:latest scan dir:/workspace --output spdx-json > sbom-new.json
+
+# If an existing SBOM file doesn't exist, save the new one as the baseline
+if [ ! -f sbom.json ]; then
+  echo "No existing SBOM found. Creating baseline SBOM."
+  mv sbom-new.json sbom.json
+  exit 1
+fi
+
+# Use sbomdiff to check for changes
+if ! sbomdiff sbom-new.json sbom.json; then
+  echo "SBOM has changed! Please review and commit the updated SBOM."
```

Comment:
This is probably a silly question, but how should someone review the SBOM? From what I can see it's not exactly a human-readable file; perhaps including the `sbomdiff` output in this message, or at least hinting that they might use that tool, might help them review it?

Reply:
Thanks good question - the SBOM I feel can be thought of as project data / metadata, similar to that of a lockfile (which isn't usually reviewed line by line). In my testing of this I found that the `sbomdiff` output does display from pre-commit runs if and where there are issues (it shows a table of where differences were detected). The check here works similarly to that of Black or ruff-format - the first run shows a failure and automatically fixes the issue (notifying the developer that something changed). The second run passes as the file has been amended for commit into the git history.